vignettes/using_lintr.Rmd

```diff
@@ -0,0 +1,311 @@
+---
+title: "Using lintr"
+author: "Alexander Rosenstock"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Using lintr}
+  %\VignetteEngine{knitr::rmarkdown}
+  \usepackage[utf8]{inputenc}
+---
+
+This vignette describes how to set up and configure `lintr` for use with projects or packages.
+
+## Running `lintr` on a project
+
+Checking a R project for lints can be done with three different functions:
+
+* Lint a single file using `lint()`:
+
+```r
+lint(filename = "R/bad.R")
+```
+
+* Lint a directory using `lint_dir()`:
+
+```r
+lint_dir(path = "R")
+```
+
+This will apply `lint()` to all R source files matching the `pattern` argument.
+By default, this means all `.R` files as well as `knitr` formats (e.g. `.Rmd`, `.Rnw`).
+
+`lint_dir` is vectorized over `path`, so multiple directories can be linted at the same time.
+
+* Lint all relevant directories of an R package using `lint_package()`:
+
+```r
+lint_package(path = ".")
+```
+
+This will apply `lint_dir()` to all subdirectories usually containing R code in packages:
+
+ - `R` containing the package implementation.
+ - `tests` containing test code.
+ - `inst` containing sample code or vignettes that will be installed along with the package.
+ - `vignettes` containing package vignettes.
+ - `data-raw` containing code to produce `data` files.
+
+For more information about the assumed package structure, view
+[R Packages](https://r-pkgs.org/package-structure-state.html).
+
+## Configuring linters
+
+### The `.lintr` file
+
+For R projects or R packages the most convenient method of configuration is creating a `.lintr` file in the project
+root. This file is in debian control format (`?read.dcf`) and each key is evaluated as R code by `lintr` when reading
+the settings.
+
+The settings file is searched for in the following order:
+
+ 1. If `options(lintr.linter_file)` is an absolute path, this file will be used.
+    The default for this option is `".lintr"`.
+ 2. If there is a file named like `lintr.linter_file` in the currently searched directory, this file will be used.
+ 3. Parent directories of the currently searched directory are searched for the linter file. If one is found, this file
+    will be used.
+ 4. If a linter file exists in the users `HOME` directory, this file is used.
+
+If none of the steps yield an existing linter file, no configuration is loaded from the file.
+
+### Using `options()`
+
+Settings from the `.lintr` file can be overridden using `options`. An option `option_name` is read from
```

Comment:
This sentence doesn't fully make sense... `option_name` is the key in the `.lintr` file? or the option name (`getOption(option_name)`)? I believe it's the former, I think the phrasing could be clarified...

Reply:
What I mean is `option_name: xxx` in `.lintr` is sort of equivalent to `options(lintr.option_name = xxx)`.

Reply:
How about

> Values in `options()` take precedence over those in `.lintr`. Note that the key `option_name` in the `.lintr` DCF translates to an R option `lintr.option_name`. If that... file. For example, `options(lintr.exclude = "# skip lint")` it will take precedence over `exclude: # nolint` in the `.lintr` file.